### DISCUSSION TITLE
Docker pull and project containers update

### DISCUSSION THREAD

**Discussion by nicoschi** [docksal/docksal#1667] [Q&A] [answered]
Hi there, 

I need confirmation about updates of docksal images and containers. 
On my machine I use docksal images tagged with application version only, eg:
- docksal/cli:php7.4
- docksal/cli:php8.0
- docksal/mysql:5.7
- docksal/apache:2.4

and so on. 

When I execute docker pull to update one of these image (eg. docker pull docksal/mysql:5.7) on fin start or restart of a project which uses the updated image the related container is being recreated. 

Now the question: is this container recreation supposed to not cause any data loss, in particular for containers containing db data?

Thanks

**Comment by lmakarov** [comment, marked as answer]
> is this container recreation supposed to not cause any data loss, in particular for containers containing db data?

It depends.

Data stored in [volumes](https://github.com/docksal/docksal/blob/develop/stacks/volumes-bind.yml) is preserved during container recreation (including when switching images tags).

Anything outside of volumes (e.g., manually installed OS packages) gets reset.

Specifically, db data is preserved as its stored within a [volume](https://github.com/docksal/docksal/blob/8045107335c20bd5f1549d7534dc458b7b729b33/stacks/services.yml#L71).
